src/content/services/monetizationLinkManager.ts

```diff
@@ -161,27 +162,28 @@ export class MonetizationLinkManager extends EventEmitter {
     }
   };
 
-  private getMonetizationLinkTags(root?: HTMLElement): HTMLLinkElement[] {
+  private getMonetizationLinkTags(root?: HTMLElement): Set<HTMLLinkElement> {
     if (this.isTopFrame) {
       const parentNode = root ?? this.document;
-      return Array.from(
+      return new Set(
         parentNode.querySelectorAll<HTMLLinkElement>(
           'link[rel="monetization"]',
         ),
       );
     } else {
       if (root && !root.closest('head')) {
-        return [];
+        return new Set();
       }
+
       const monetizationTag = this.document.querySelector<HTMLLinkElement>(
         'head link[rel="monetization"]',
       );
-      return monetizationTag ? [monetizationTag] : [];
+      return new Set(monetizationTag ? [monetizationTag] : []);
     }
   }
 
   /** @throws never throws */
-  private async checkLink(link: HTMLLinkElement) {
+  private async getLinkDetails(link: HTMLLinkElement) {
```

Comment:
Let's avoid renaming so we can get a clearer diff of what we're doing in this PR. Can do renaming as a follow up, so each merged commit only does one thing it aimed to do.